easybuild/easyblocks/l/libqglviewer.py

```diff
@@ -49,10 +50,26 @@ def configure_step(self):
     def sanity_check_step(self):
         """Custom sanity check for libQGLViewer."""
         shlib_ext = get_shared_lib_ext()
+        """
+        From version 2.8.0 onwards qt version also gets added to the lib file names.
+        """
+        if LooseVersion(self.version) < LooseVersion("2.8.0"):
+            custom_paths = {
+                'files': [('lib/libQGLViewer.prl', 'lib64/libQGLViewer.prl'),
+                          ('lib/libQGLViewer.%s' % shlib_ext, 'lib64/libQGLViewer.%s' % shlib_ext)],
+                'dirs': ['include/QGLViewer'],
+            }
+        else:
+            addition = ''
+            for dep in self.cfg.dependencies():
+                if (dep['name'][0] == 'Q' and dep['name'][1] == 't'):
+                    addition = dep['name'].lower()
+                    addition = '-' + addition
```

Comment:
I think this looks a bit cryptic and seems a bit fragile considering things like `QtPy` `Qtconsole`  etc.
I'd say just go with hardcoding Qt5 and 6 for now. We'll worry about generalizing it in 10 years when Qt7 comes.
```suggestion
            for qt_dep in ['Qt5', 'Qt6']:
                if get_software_root('Qt5'):
                    addition = qt_dep.lower()
                    break
            else:
                raise EasyBuildError("Missing Qt5 or Qt6 dependency")
```